### PR TITLE
Copy functionality

### PR DESCRIPTION
**Pull request by german-e-mas**
Allows the user to copy Articles. Their corresponding comments are copied as well.

Modifying or deleting the original does not changes the copy.

ptal @basicNew

### CODE REVIEW

File: app/controllers/articles_controller.rb [hunk 1]
```diff
@@ -41,6 +41,22 @@ def destroy
 		redirect_to articles_path
 	end
 
+	def copy
+		# Get the corresponding Article and save a copy.
+		@article = Article.new
```
Comment (basicNew) [inline app/controllers/articles_controller.rb]: There is no need to do this, as you are overriding whatever contents where in the variable in tne next statement. Remember that variables in Ruby are not typed. Same goes for: 

```
new_comment = Comment.new
new_comment = comment.d
```

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -41,6 +41,22 @@ def destroy
...
+		@article.text.prepend("[Copy] ")
+		@article.save
+		# Copy the comments in the original Article.
+		Comment.where(article_id: params[:id]).each do |comment|
+			new_comment = Comment.new
+			new_comment = comment.dup
+			new_comment.article_id = @article.id
+			new_comment.save
+		end
+		redirect_to @article
```
Comment (basicNew) [inline app/controllers/articles_controller.rb]: Nice one!

File: app/controllers/articles_controller.rb (hunk 1 above)
Comment (german-e-mas) [inline app/controllers/articles_controller.rb]: You are right. I tested it and this line is not necessary at all. Will change these lines!

**Comment by german-e-mas** [comment]
I removed the lines containing the `.new` method. Tested the changes and everything works smoothly.

ptal again @basicNew

**Comment by german-e-mas** [comment]
Merging. Thanks for the review @basicNew

**Pull request merged**